standard/stream_links.lua

```diff
@@ -93,11 +95,101 @@ function StreamLinks.processStreams(forwardedInputArgs)
 
 				streamValue = StreamLinks.resolve(lookUpPlatform, streamValue)
 			end
+
+			if FeatureFlag.get('new_stream_format') then
+				local key = StreamLinks.StreamKey(platformName):toString()
+				streams[key] = streamValue
+			end
 			streams[platformName] = streamValue
 		end
 	end
 
 	return streams
 end
 
+--- StreamKey Class
+-- Contains the triplet that makes up a stream key
+-- [platform, languageCode, index]
+StreamLinks.StreamKey = Class.new(
+	function (self, ...)
+		self:new(...)
+	end
+)
+local StreamKey = StreamLinks.StreamKey
+
+function StreamKey:new(tbl, languageCode, index)
+	local platform
+	-- Input is another StreamKey - Make a copy
+	if StreamKey._isStreamKey(tbl) then
+		platform = tbl.platform
+		languageCode = tbl.languageCode
+		index = tbl.index
+	-- Input is another table, assume format is {platform, languageCode, index}
+	elseif type(tbl) == 'table' then
+		platform, languageCode, index = unpack(tbl)
+	-- All three parameters are supplied
+	elseif languageCode and index then
+		platform = tbl
+	elseif type(tbl) == 'string' then
+		local components = mw.text.split(tbl, '_', true)
+		-- Input is in legacy format (eg. twitch2)
+		if #components == 1 then
+			platform, index = self:_fromLegacy(tbl)
+			languageCode = 'en'
+		-- Input is a StreamKey in string format
+		elseif #components == 3 then
+			platform, languageCode, index = unpack(components)
+		end
+	end
+
+	self.platform = platform
+	self.languageCode = languageCode
+	self.index = tonumber(index)
+	self:_isValid()
+	self.languageCode = self.languageCode:lower()
+end
+
+function StreamKey:_fromLegacy(input)
+	for _, platform in pairs(StreamLinks.countdownPlatformNames) do
+		-- The intersection of values in countdownPlatformNames and keys in streamPlatformLookupNames
+		-- are not valid platforms.
+		if not StreamLinks.streamPlatformLookupNames[platform] then
+			-- Check if this platform matches the input
+			if string.find(input, platform, 1, true) then
+				local index
+				-- If the input is longer than the platform, there's an index at the end
+				if #input > #platform then
+					index = tonumber(input:sub(#platform + 1))
+				else
+					index = 1
+				end
+				return platform, index
+			end
+		end
+	end
+end
+
+function StreamKey:toString()
+	return self.platform .. '_' .. self.languageCode .. '_' .. self.index
+end
+
+function StreamKey:toLegacy()
+	return self.platform .. (self.index > 1 and self.index or '')
```

Comment:
There are a lot of null checks when setting `self.index`, do we need it here or can we assert not null?

Reply:
This is not a null check, rather, we want don't to suffix it if the `index` is `1`. Otherwise we'd output `twitch1` when the expected output is `twitch`.

Reply:
Change the function and added a comment for clarity.

Reply:
My comment was about this part: `self.index or ''`. That's now been resolved